### DISCUSSION TITLE
How to generate different kind of validation annotations for different type of List conditions in graphQL schema?

### DISCUSSION THREAD

**Discussion by siva54** [kobylynskyi/graphql-java-codegen#955] [Q&A] [answered]
Hi, 
I was trying to generate the schema with a List type and came across some issues. As per the schema we have different use cases for a not-null condition for a list,
![image](https://user-images.githubusercontent.com/1768164/151841366-82ab4705-a2d4-4308-9600-e173aebb8449.png)
In my project, I'm using a couple of cases too such as,
[Int!] - This means that the list can be null but cannot have null objects in the list.
[Int]! - This means that the list cannot be null but can have null objects in the list.
When using codegen to generate POJOs, for types with
[Int!]
[Int]!

In both cases, the field is being annotated with,
@javax.validation.constraints.NotNull

This basically means that the List cannot be null. This is incorrect for the type [Int!] and caused some trouble. As of now, I replaced all the NotNull to instead use a PlaceboValidation annotation of mine that does nothing. How do I overcome this and make sure to ensure that the POJO created works as expected?

**Comment by siva54** [comment]
> Hi, I was trying to generate the schema with a List type and came across some issues. As per the schema we have different use cases for a not-null condition for a list, ![image](https://user-images.githubusercontent.com/1768164/151841366-82ab4705-a2d4-4308-9600-e173aebb8449.png) In my project, I'm using a couple of cases too such as, [Int!] - This means that the list can be null but cannot have null objects in the list. [Int]! - This means that the list cannot be null but can have null objects in the list. When using codegen to generate POJOs, for types with [Int!] [Int]!
> 
> In both cases, the field is being annotated with, @javax.validation.constraints.NotNull
> 
> This basically means that the List cannot be null. This is incorrect for the type [Int!] and caused some trouble. As of now, I replaced all the NotNull to instead use a PlaceboValidation annotation of mine that does nothing. How do I overcome this and make sure to ensure that the POJO created works as expected?

Hi @kobylynskyi . Any idea on this one?

**Reply by kobylynskyi** [reply]
This looks like a bug in _AnnotationsMapper_ class.
I will do my best to try looking at it soon, but can't promise anything because of the load at my other job.
Thank you for understanding.

**Reply by siva54** [reply]
Thank you, @kobylynskyi .

**Comment by kobylynskyi** [comment, marked as answer]
@siva54 Addressed this issue in #960 and it will be released as part of 5.4.1.

**Reply by siva54** [reply]
Thank you @kobylynskyi

**Reply by kobylynskyi** [reply]
Created 5.4.1 release PR: #980
Sorry for addressing it so long.
